pathway-analysis/kegg_ortholog_mapping.Rmd

```diff
@@ -0,0 +1,252 @@
+---
+title: "KEGG pathways: mapping to mouse orthologs with `hcop`"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+author: J. Taroni for ALSF CCDL
+date: 2019
+---
+
+## Background
+
+In this module, we use QuSAGE ([Yaari et al. _NAR_. 2013.](https://doi.org/10.1093/nar/gkt660))
+for pathway analysis (implemented in the [`qusage` bioconductor package](https://bioconductor.org/packages/release/bioc/html/qusage.html)).
+
+`qusage` allows you to read in gene sets that are in the [GMT format](http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29).
+
+[MSigDB](http://software.broadinstitute.org/gsea/msigdb) offers genesets in this format.
+[Curated gene sets](http://software.broadinstitute.org/gsea/msigdb/collections.jsp#C2) 
+such as [KEGG](https://www.genome.jp/kegg/) are a good starting point for any pathway analysis.
+
+However, MSigDB only distributes human pathways.
+If we want to use KEGG Pathways with another species without going through
+[KEGG Orthology](https://www.genome.jp/kegg/ko.html), we need to map to
+orthologs ourselves.
+
+We'll use the [`hcop` package](https://github.com/stephenturner/hcop) to do 
+this.
+If you're looking for a little bit more background information (like if you
+run into trouble installing `hcop`), check out the notebook in our
+[`ortholog-mapping`](https://github.com/AlexsLemonade/refinebio-examples/tree/master/ortholog-mapping) module.
+
+## Setup
+
+Package installation
+
+```{r}
+# need read.gmt functionality from qusage
+if (!("qusage" %in% installed.packages())) {
+  BiocManager::install("qusage", update = FALSE)
+}
+
+# we need devtools in order to install the hcop package we will use to
+# do the ortholog mapping
+if (!("devtools" %in% installed.packages())) {
+  install.packages("devtools")
+}
+
+# this installs a specific version of hcop
+# we pass the commit hash to the ref argument
+devtools::install_github("stephenturner/hcop", 
```

Comment:
Should there be an `if` statement for this too? I realize that this package isn't as ubiquitous as `devtools` but if they ran the `ortholog-mapping` notebook, they might have already installed it.

Reply:
I did not use an `if` here because we're installing a _specific commit_. If I didn't pass a commit hash to `ref`, I would have used an `if` statement. Running this when you have the same specific version install just tells you that it has not been updated since the reference you're passing it.

Reply:
Ah. Okay. Cool. Sounds good.